### PR TITLE
bf: ZENKO-1155 remove incremental update

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

**Comment by alexanderchan-scality** [comment]
@bert-e wait

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 1]
```diff
@@ -551,7 +551,9 @@ class MongoClientInterface {
         const { objName, versionId, collectionName } = params;
         if (isUserBucket(collectionName)) {
             if (isUpdate) {
+                const findId = new RegExp(objName);
```
Comment (jonathan-gramain) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: Why the need for a regexp?

Comment (alexanderchan-scality) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js]: What this code attempts to do is to find objects with a specific version id; in this case, it is attempting to find the master(placeholder), an entry that doesn't  the versioned id appended, and the corresponding version id. So, the regex will looking for entry-ids that match the regex.

Comment (vrancurel) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js]: yes but there is a chance it translates to a SCAN as well

**Review by vrancurel — CHANGES_REQUESTED** [review]

**Review by vrancurel — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Review by bennettbuchanan — APPROVED** [review]

**Pull request merged**